runtime/bcverify/j9bcverify.tdf

```diff
@@ -91,7 +91,7 @@ TraceExit=Trc_RTV_unlinkClassLoadingConstraints_Exit NoEnv Overhead=1 Level=1 Te
 
 TraceEvent=Trc_RTV_checkClassLoadingConstraintForName Overhead=1 Level=3 Template="checkClassLoadingConstraintForName - Checking constraints between %p and %p for %.*s"
 
-TraceAssert=Assert_RTV_validateClassLoadingConstraints Overhead=2 Level=5 NoEnv Assert="validateArgs(P1, P2, P3, P4, P5, P6), 1"
+TraceAssert=Assert_RTV_validateClassLoadingConstraints Overhead=2 Level=5 NoEnv Assert="validateArgs(P1, P2, P3, P4, P5, P6, P7, P8), 1"
```

Comment:
This trace point is level 5, which is not enabled by default. 
We cannot change an existing trace point like this. It could cause problems parsing snap trace file generated by an older build that does not have this change. 

Instead of changing this trace point, maybe you can add a new function like the following to set `copyUTF1` and `copyUTF2`
```
static void 
findNamesInSegments(validateArgs (J9VMThread *vmThread, U_8 *name1, U_8 *name2, UDATA *name1Found, UDATA *name2Found) 
{
	J9MemorySegment *seg = vmThread->javaVM->classMemorySegments->nextSegment;
	while (seg) {
		if (!(*name1Found) 
                    && (seg->heapBase <= name1) 
                    && (seg->heapTop >= name1)
                  ) {
			*name1Found = TRUE;
		}
		/* same for name2 */
                 /* break if both names are found in the segment */
		seg = seg->nextSegment;
	}
}
```

Reply:
I just noticed this trace point is level 2, so searching the classname is the class segment will have performance impact. There is a better way to fix this. Looking at the caller of `j9bcv_checkClassLoadingConstraintsForSignature()`, if the `sig1` and `sig2` are from the J9ROM** macro, then they are in the class segment and don't need to be copied.

The only case that needs to be copied is `lookupSig` here:
https://github.com/eclipse-openj9/openj9/blob/d1adf51763025c486fbfd525fab01c95bafcf997/runtime/vm/lookupmethod.c#L239-L240

We could change `j9bcv_checkClassLoadingConstraintsForSignature()` to pass in additional parameters to whether copy the strings.